### PR TITLE
sharing vs sn changes

### PR DESCRIPTION
**Pull request by michaelkirk**
PTAL @charlesmchen

### CODE REVIEW

File: Signal/src/ViewControllers/DebugUI/DebugUISessionState.m [hunk 1]
```diff
@@ -81,56 +77,6 @@ - (nullable OWSTableSection *)sectionForThread:(nullable TSThread *)threadParame
...
-                                                      handler:^(UIAlertAction *_Nonnull action) {
-                                                          NSData *identityKey =
-                                                              [identityManger identityKeyForRecipientId:recipientId];
-                                                          [[OWSIdentityManager sharedManager]
-                                                               setVerificationState:OWSVerificationStateVerified
-                                                                        identityKey:identityKey
-                                                                        recipientId:recipientId
-                                                              isUserInitiatedChange:NO];
-                                                      }]];
-    [alertController addAction:[UIAlertAction actionWithTitle:@"No Longer Verified"
```
Comment (michaelkirk) [inline Signal/src/ViewControllers/DebugUI/DebugUISessionState.m]: This skips some important machinery - if you want someone to be marked as no-longer-verified, first verify them in the fingerprint controller, then use the debug UI to toggle a SN change.

File: SignalMessaging/attachments/AttachmentApprovalViewController.swift [hunk 2]
```diff
@@ -231,36 +231,53 @@ public class AttachmentApprovalViewController: OWSViewController, CaptioningTool
         // Toolbar flickers in and out if there are errors
         // and remains visible momentarily after share extension is dismissed.
         // It's easiest to just hide it at this point since we're done with it.
-        self.bottomToolbar.isUserInteractionEnabled = false
-        self.bottomToolbar.isHidden = true
+        shouldAllowAttachmentViewResizing = false
```
Comment (michaelkirk) [inline SignalMessaging/attachments/AttachmentApprovalViewController.swift]: Presenting/dismissing alerts can cause the view controller to become/resign first responder, which has implication for when the keyboard is presented which in turn affects the zoom status of the attachment view.

So we make sure to not fire any more of those animations after the send button has been hit.

File: Signal/translations/en.lproj/Localizable.strings [hunk 3]
```diff
@@ -1540,6 +1540,9 @@
 /* action sheet item */
 "SHARE_ACTION_TWEET" = "Twitter";
 
+/* alert body when sharing filefailed because of untrusted/changed identity keys */
+"SHARE_EXTENSION_FAILED_SENDING_BECAUSE_UNTRUSTED_IDENTITY_KEY" = "Your safety number with %@ has recently changed. You may wish to verify it in the main app before resending.";
```
Comment (charlesmchen) [inline Signal/translations/en.lproj/Localizable.strings]: TIOLI - I tend to use a `_FORMAT` suffix for l10n strings that should be formatted.

File: SignalMessaging/attachments/AttachmentApprovalViewController.swift [hunk 4]
```diff
@@ -231,36 +231,53 @@ public class AttachmentApprovalViewController: OWSViewController, CaptioningTool
...
+    // so we sometimes allows resizing the attachment.
+    private var shouldAllowAttachmentViewResizing: Bool = true
 
-                    let originalHeight = self.scrollView.bounds.size.height
-
-                    // Position the new scaled item to be centered with respect
-                    // to it's new size.
-                    let heightDelta = originalHeight * (1 - kScaleFactor)
-                    let translate = CGAffineTransform(translationX: 0, y: -heightDelta / 2)
+    enum AttachmetViewScale {
```
Comment (charlesmchen) [inline SignalMessaging/attachments/AttachmentApprovalViewController.swift]: s/AttachmetViewScale/AttachmentViewScale/

File: SignalMessaging/attachments/AttachmentApprovalViewController.swift [hunk 5]
```diff
@@ -231,36 +231,53 @@ public class AttachmentApprovalViewController: OWSViewController, CaptioningTool
...
+        case fullsize, compact
+    }
```
Comment (charlesmchen) [inline SignalMessaging/attachments/AttachmentApprovalViewController.swift]: IMO all class properties (e.g. `shouldAllowAttachmentViewResizing `) and inner types (e.g. `AttachmetViewScale`) belong at the top of the class, before any methods.

File: SignalMessaging/attachments/SharingThreadPickerViewController.m [hunk 6]
```diff
@@ -233,44 +216,183 @@ - (void)tryToSendAttachment:(SignalAttachment *)attachment fromViewController:(U
...
-    UIAlertAction *retryAction =
-        [UIAlertAction actionWithTitle:[CommonStrings retryButton]
-                                 style:UIAlertActionStyleDefault
-                               handler:^(UIAlertAction *action) {
-                                   [self tryToSendAttachment:attachment fromViewController:fromViewController];
-                               }];
-    [failureAlert addAction:retryAction];
-
-    [fromViewController presentViewController:failureAlert animated:YES completion:nil];
+    if (error.code == OWSErrorCodeUntrustedIdentity) {
```
Comment (charlesmchen) [inline SignalMessaging/attachments/SharingThreadPickerViewController.m]: Don't you need to check the error domain too?

File: Signal/translations/en.lproj/Localizable.strings (hunk 3 above)
Comment (michaelkirk) [reply Signal/translations/en.lproj/Localizable.strings]: Done!

File: SignalMessaging/attachments/AttachmentApprovalViewController.swift (hunk 4 above)
Comment (michaelkirk) [reply SignalMessaging/attachments/AttachmentApprovalViewController.swift]: Thanks.

File: SignalMessaging/attachments/AttachmentApprovalViewController.swift (hunk 5 above)
Comment (michaelkirk) [reply SignalMessaging/attachments/AttachmentApprovalViewController.swift]: Done.

File: SignalMessaging/attachments/SharingThreadPickerViewController.m (hunk 6 above)
Comment (michaelkirk) [reply SignalMessaging/attachments/SharingThreadPickerViewController.m]: Done!

**Pull request merged**